lib/tasks/widgets/new_task_dialog.dart

```diff
@@ -2,20 +2,30 @@ import 'package:flutter/material.dart';
 import 'package:studypoints/tasks/data/task.dart';
 
 class NewTaskDialog extends StatefulWidget {
+  final Task task;
+  NewTaskDialog({this.task});
+
   @override
   _NewTaskDialogState createState() => _NewTaskDialogState();
 }
 
 class _NewTaskDialogState extends State<NewTaskDialog> {
-  int priority = 0;
+  int priority;
   DateTime dueDate;
-  TextEditingController titleController = TextEditingController();
+  TextEditingController titleController;
   List<TextEditingController> subtaskControllers = [];
   List<Widget> subtaskInputs = [];
   final GlobalKey<FormState> _formKey = GlobalKey<FormState>();
 
   @override
   Widget build(BuildContext context) {
+    titleController ??= TextEditingController(text: widget.task?.title);
+    if (widget.task != null) {
+      priority ??= widget.task.priority;
```

Comment:
Instead of checking if `widget.task` is null, try something along the lines of `priority = widget.task?.priority ?? 0`